lib/modem_jwt/modem_jwt.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <string.h>
+#include <zephyr.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <modem/at_cmd.h>
+#include <modem/modem_jwt.h>
+
+#define JWT_CMD_TEMPLATE "AT%%JWT=%d,%u,\"%s\",\"%s\",%d,%d\r\n"
+#define MAX_INT_PRINT_SZ 11
+#define JWT_CMD_PRINT_INT_SZ (MAX_INT_PRINT_SZ * 4)
+
+/* Minimum JWT response is ~420 bytes.
+ * With very long (128 bytes) values for sub and aud claims
+ * response is ~820.
+ */
+BUILD_ASSERT(CONFIG_MODEM_JWT_MAX_LEN <= CONFIG_AT_CMD_RESPONSE_MAX_LEN,
+	     "Max JWT length exceeds max AT cmd response length");
+
+static int parse_jwt_at_cmd_resp(char *const jwt_resp)
+{
```

Comment:
Is this parsing something that Jan Tore is adding to the lte link control library?

Reply:
i do not know
@jtguggedal ? :eyes:

Reply:
No plans to add JWT support to the link controller, so I think we'll keep the JWT stuff in this library unless there's a use case for combining it with link control somehow.